pkgs/development/compilers/ghc/cabal-paths.patch

```diff
@@ -0,0 +1,40 @@
+diff --git a/Cabal/Distribution/Simple/Build/PathsModule.hs b/Cabal/Distribution/Simple/Build/PathsModule.hs
+index 5e660e8d6..22ce5b4fe 100644
+--- a/libraries/Cabal/Cabal/Distribution/Simple/Build/PathsModule.hs
++++ b/libraries/Cabal/Cabal/Distribution/Simple/Build/PathsModule.hs
+@@ -32,6 +32,7 @@ import Distribution.Pretty
+ import Distribution.Version
+ 
+ import System.FilePath ( pathSeparator )
++import Data.List ( isPrefixOf )
+ 
+ -- ------------------------------------------------------------
+ -- * Building Paths_<pkg>.hs
+@@ -82,8 +83,8 @@ generatePathsModule pkg_descr lbi clbi =
+         pragmas++
+         "module " ++ prettyShow paths_modulename ++ " (\n"++
+         "    version,\n"++
+-        "    getBinDir, getLibDir, getDynLibDir, getDataDir, getLibexecDir,\n"++
+-        "    getDataFileName, getSysconfDir\n"++
++        nixShouldEmitPaths ("    getBinDir, getLibDir, getDynLibDir, getDataDir, getLibexecDir,\n"++
```

Comment:
This is a too agressive approach, since it breaks the use of data files with a separate bin output completely. This can work if the data files are installed to a separate output.

Ideally we would check for every path individually whether to emit the relevant bindings using the following condition: Emit if bindir is below prefix or this path is not below prefix.

E.g.:

* libdir etc. wouldn't be emitted if we have a separate bindir because it is below prefix
* e. g. datadir would be emitted if it is in a separate output and we have a separate bindir, but not if it is below prefix (which usually means we don't use the data dir at all)
* if bindir is below prefix, emit everything (like now)

Reply:
Okay, I've pushed up a new version that should have implemented this.  It feels a bit hack-y and it's not the most beautiful Haskell but it works!